src/hazelcore/HTyp.re

```diff
@@ -1,9 +1,12 @@
 open Sexplib.Std;
 
+[@deriving sexp]
+type hole_provenance = option(unit);
```

Comment:
My interpretation was that Some(()) meant that it was a type hole (i.e. from a syntactic location in the code -- in the future, the Some case would contain the provenance data itself) and None was that it was an internally generated unknown type, but that seems to be backwards from what you implemented e.g. in Action_Exp? (and perhaps other places, haven't read through it all yet...)

Reply:
My recollection is that this was the way I did it initially, and then you told me it should be the other way around. I've definitely already done a bulk-replace to get them in the current correspondence.

Reply:
Unless these are inconsistencies you're pointing out versus a systematic issue?

Reply:
My recollection was that we had settled on the interpretation above, which seems to make sense to me. Apologies if it was unclear though. It's probably worth explicitly documenting what Some() and None mean on this line and then make sure that is how they are used. If it is systematically swapped everywhere then that should be a simple find/replace to fix it. But I can also do a more careful pass through the key syn/ana branches once that is settled...